source/common/router/router.cc

```diff
@@ -797,12 +797,13 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
         continue;
       }
       auto shadow_headers = Http::createHeaderMap<Http::RequestHeaderMapImpl>(*shadow_headers_);
+      auto sampled = shadow_policy.traceSampled() ? absl::nullopt : absl::optional<bool>(false);
```

Comment:
not sure if this should be inlined or if we want to move this to a private method or something else